src/tiled/variantpropertymanager.cpp

```diff
@@ -126,6 +135,32 @@ QString VariantPropertyManager::valueText(const QtProperty *property) const
         QVariant value = mValues[property].value;
         int typeId = propertyType(property);
 
+        if (typeId == objectIdTypeId()) {
+            ObjectId id = value.value<ObjectId>();
+
+            if (MapDocument *document = DocumentManager::instance()->currentDocument()) {
```

Comment:
Hmm, this may be a little problematic. There is no guarantee that the current document is the document to which the given `ObjectId` value is referring. Most of the time it will be true, but for example in the Tile Collision Editor, the objects have their own ID space (though it's a little hacky). I think the only real solution there would be to make the `ObjectId` contain also a `Map*`, but actually that won't work for the tile collision editor case since it uses a bare `ObjectGroup` instance. And if we add an `ObjectGroup*` to the `ObjectId` then we'd have to go and keep that up to date, which would be annoying as well.

What are you thoughts on this?